src/json-validator.cpp

```diff
@@ -186,64 +186,98 @@ void validate_boolean(const json & /*instance*/, const json &schema, const std::
 	validate_type(schema, "boolean", name);
 }
 
-void validate_numeric(const json &schema, const std::string &name, double value)
+template <class T>
+bool violates_numeric_maximum(T max, T value, bool exclusive)
 {
+	if (exclusive)
+		return value > max;
+
+	return value >= max;
+}
+
+template <class T>
+bool violates_numeric_minimum(T min, T value, bool exclusive)
+{
+	if (exclusive)
+		return value < min;
+
+	return value <= min;
+}
+
```

Comment:
These two functions are wrong actually regarding the exclusive argument: If exclusive is true, the value violates the schema if it is inclusive regarding the min-value. So somewhere things need to be swapped. Either `violates` becomes `accept` or the `if` branches are swapped.

Reply:
Ah, you are correct. There was not a bug in the comparison logic in the first place, just that it wasn't examining whether the field was true or false.